### PR TITLE
Add conformance tests for lite runtime

### PR DESCRIPTION
**Pull request by pkwarren**
Update the conformance/google-javalite project to run tests with the lite runtime.

### CODE REVIEW

File: conformance/buf.gen.yaml [hunk 1]
```diff
@@ -20,6 +20,10 @@ plugins:
   - plugin: connect-kotlin
     out: google-javalite/build/generated/sources/bufgen
     path: ./protoc-gen-connect-kotlin/build/install/protoc-gen-connect-kotlin/bin/protoc-gen-connect-kotlin
+    opt:
```
Comment (pkwarren) [inline conformance/buf.gen.yaml:23]: Updated the lite tests to use generated code with the same args as the full runtime tests.

File: conformance/google-java/src/test/kotlin/com/connectrpc/conformance/java/ConformanceTest.kt [hunk 2]
```diff
@@ -44,120 +40,27 @@ import kotlinx.coroutines.async
...
-            httpClient = ConnectOkHttpClient(client),
-            ProtocolClientConfig(
-                host = host,
-                serializationStrategy = GoogleJavaProtobufStrategy(),
-                networkProtocol = protocol,
-                requestCompression = RequestCompression(10, GzipCompressionPool),
-                compressionPools = listOf(GzipCompressionPool),
-            ),
-        )
+        init(GoogleJavaProtobufStrategy())
```
Comment (pkwarren) [inline conformance/google-java/src/test/kotlin/com/connectrpc/conformance/java/ConformanceTest.kt:63]: This is the only real change between the two clients.

File: conformance/google-javalite/build.gradle.kts [hunk 3]
```diff
@@ -22,9 +22,9 @@ sourceSets {
 
 dependencies {
     implementation(libs.kotlin.coroutines.core)
-    implementation(libs.protobuf.kotlin)
+    implementation(libs.protobuf.kotlinlite)
     implementation(project(":conformance:common"))
-    implementation(project(":extensions:google-java"))
+    implementation(project(":extensions:google-javalite"))
```
Comment (pkwarren) [inline conformance/google-javalite/build.gradle.kts:27]: These dependencies weren't correct before.

File: conformance/google-javalite/src/test/kotlin/com/connectrpc/conformance/javalite/ConformanceTest.kt [hunk 4]
```diff
@@ -0,0 +1,704 @@
...
+import org.assertj.core.api.Assertions.fail
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.Parameterized
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+@RunWith(Parameterized::class)
+class ConformanceTest(
```
Comment (pkwarren) [inline conformance/google-javalite/src/test/kotlin/com/connectrpc/conformance/javalite/ConformanceTest.kt:53]: I don't like the fact that this is a duplicate of the non-lite test (except for the serialization strategy), but I don't know a better way to share logic with different dependencies.

**Review by smaye81 — APPROVED** [review]

**Pull request merged**